### PR TITLE
fix(curriculum): add `for` attribute to `label` elements in registration-form project

### LINKED ISSUE #46593: Accessibility issue with labels in Registration Form course
### Describe the Issue

There are two related issues:

- The label for the file input does not use a `for` attribute and this is causing the NVDA screen reader on Windows and ORCA screen reader on linux to not announce the label text when the user tabs to the input. I have a [codepen demo](https://codepen.io/bbsmooth/pen/WNMBjVK) to illustrate this.
- It is considered a [best practice to use a `for` attribute on a label,](https://www.w3.org/WAI/tutorials/forms/labels/) even when the input is embedded in the label itself, because some screen readers may not read the label correctly without the `for` attribute (as demonstrated in the issue above). This is also noted in [step 30 of the Accessibility Quiz course](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-30).

All of the inputs on the form in this course are embedded in a `label` and none of the labels have a `for` attribute.

### Affected Page

All steps of the Registration Form course that have inputs embedded in labels.

### Your code

Current HTML for file input:

```
<label>Upload a profile picture: <input type="file" name="file" /></label>
```

### Expected behavior

Should be something like:

```
<label for="file-upload">Upload a profile picture: <input id="file-upload" type="file" name="file" /></label>
```

### Screenshots

_No response_

### System

- Device: Desktop
- OS: Windows 11
- Browser: Firefox
- Version: 101
- Screen Reader: NVDA
- Version: 2022.1

### Additional context

I'm only bringing this up because I thought it might be a good opportunity to teach campers that valid HTML isn't always as accessible as it could be and best practices should be followed because they help eliminate these types of edge cases.

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47069
Closes #46593

I ended creating additional steps in which to add the ids and for attributes, otherwise I had no idea were to add the tests...
I have added three new steps, 17, 28, 37, in which the camper will have to manually add ids and for attributes for all the labels, and changed seed code and solutions appropriately on all the following ones. I used #47069 as a starting point.

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fab9f17fa294054b74228c.md [hunk 1]
```diff
@@ -48,18 +48,18 @@ assert.equal(document.querySelector('fieldset:nth-child(3) > label > input')?.ty
...
-        <label>
-          <input type="checkbox" required /> I accept the
-            <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
-        </label>
+        <label for="personal-account"><input id="personal-account" type="radio" name="account-type" /> Personal Account</label>
+        <label for="business-account"><input id="business-account" type="radio" name="account-type" /> Business Account</label>
+        <label for="terms-and-conditions">
+          <input id="terms-and-conditions" type="checkbox" required /> I accept the
+				    <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
+			  </label>
```
Comment (ieahleen) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fab9f17fa294054b74228c.md:62]: need to fix formatting

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fabf0dd4959805dbae09e6.md [hunk 2]
```diff
@@ -62,17 +62,18 @@ assert.equal(document.querySelector('fieldset:nth-child(3) > label:nth-child(2)
...
-        <label><input type="radio" name="account-type" /> Business Account</label>
-        <label>
-          <input type="checkbox" required /> I accept the <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
-        </label>
+        <label for="personal-account"><input id="personal-account" type="radio" name="account-type" /> Personal Account</label>
+        <label for="business-account"><input id="business-account" type="radio" name="account-type" /> Business Account</label>
+        <label for="terms-and-conditions">
+          <input id="terms-and-conditions" type="checkbox" required /> I accept the
+				    <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
+			  </label>
```
Comment (ieahleen) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fabf0dd4959805dbae09e6.md:76]: same here

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fac4095512d3066053d73c.md [hunk 3]
```diff
@@ -42,17 +42,18 @@ assert.equal(document.querySelectorAll('fieldset:nth-child(3) > select > option'
...
-        <label><input type="radio" name="account-type" /> Personal Account</label>
-        <label><input type="radio" name="account-type" /> Business Account</label>
-        <label>
-          <input type="checkbox" required /> I accept the <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
-        </label>
+        <label for="personal-account"><input id="personal-account" type="radio" name="account-type" /> Personal Account</label>
+        <label for="business-account"><input id="business-account" type="radio" name="account-type" /> Business Account</label>
+        <label for="terms-and-conditions">
+          <input id="terms-and-conditions" type="checkbox" required /> I accept the
+				    <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
```
Comment (ieahleen) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fac4095512d3066053d73c.md:55]: and here

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fac56271087806def55b33.md [hunk 4]
```diff
@@ -52,17 +52,18 @@ assert.equal(document.querySelectorAll('fieldset > label > select > option')?.le
...
-        <label><input type="radio" name="account-type" /> Personal Account</label>
-        <label><input type="radio" name="account-type" /> Business Account</label>
-        <label>
-          <input type="checkbox" required /> I accept the <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
-        </label>
+        <label for="personal-account"><input id="personal-account" type="radio" name="account-type" /> Personal Account</label>
+        <label for="business-account"><input id="business-account" type="radio" name="account-type" /> Business Account</label>
+        <label for="terms-and-conditions">
+          <input id="terms-and-conditions" type="checkbox" required /> I accept the
+				    <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
```
Comment (ieahleen) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fac56271087806def55b33.md:65]: and here

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fac8d7fdfaee0796934f20.md [hunk 5]
```diff
@@ -66,17 +66,18 @@ assert.equal(document.querySelectorAll('fieldset:nth-child(3) > label:nth-child(
...
-        <label><input type="radio" name="account-type" /> Personal Account</label>
-        <label><input type="radio" name="account-type" /> Business Account</label>
-        <label>
-          <input type="checkbox" required /> I accept the <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
-        </label>
+        <label for="personal-account"><input id="personal-account" type="radio" name="account-type" /> Personal Account</label>
+        <label for="business-account"><input id="business-account" type="radio" name="account-type" /> Business Account</label>
+        <label for="terms-and-conditions">
+          <input id="terms-and-conditions" type="checkbox" required /> I accept the
+				    <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
```
Comment (ieahleen) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fac8d7fdfaee0796934f20.md:79]: and here

**Comment by ieahleen** [comment]
I solved the merge conflicts in the wrong way, will be back

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fab8367d35de04e5cb7929.md [hunk 6]
```diff
@@ -1,8 +1,8 @@
 ---
 id: 60fab8367d35de04e5cb7929
-title: Step 27
```
Comment (Sembauke) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fab8367d35de04e5cb7929.md:3]: We are going from step 27 to 29 is this correct?

Comment (ieahleen) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fab8367d35de04e5cb7929.md:3]: do you mean as the change? there is also a new step 17 so yes

Comment (ieahleen) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fab8367d35de04e5cb7929.md:3]: the name file changes were all automatic

**Review by Sembauke — DISMISSED** [review]

**Comment by ShaunSHamilton** [comment]
Blocking for now, I have requested changes, but cannot make comments in the Files tab at the moment.

**Comment by ieahleen** [comment]
@ShaunSHamilton I will wait for your review

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+The third `input` element should have an `id` of `email`.
+
+```js
+assert(document.querySelectorAll('input')[2].matches('#email'))
+```
+
+The fourth `input` element should have an `id` of `new-password`.
+
+```js
+assert(document.querySelectorAll('input')[3].matches('#new-password'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md:37]:
```suggestion
assert(document.querySelectorAll('input')?.[3]?.matches('#new-password'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+The first `label` element should have a `for` attribute with a value of `first-name`.
+
+```js
+assert(document.querySelectorAll('label')[0].matches('label[for="first-name"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md:43]:
```suggestion
assert(document.querySelectorAll('label')?.[0]?.matches('label[for="first-name"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+The second `label` element should have a `for` attribute with a value of `last-name`.
+
+```js
+assert(document.querySelectorAll('label')[1].matches('label[for="last-name"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md:49]:
```suggestion
assert(document.querySelectorAll('label')?.[1]?.matches('label[for="last-name"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+The third `label` element should have a `for` attribute with a value of `email`.
+
+```js
+assert(document.querySelectorAll('label')[2].matches('label[for="email"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md:55]:
```suggestion
assert(document.querySelectorAll('label')?.[2]?.matches('label[for="email"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+The fourth `label` element should have a `for` attribute with a value of `new-password`.
+
+```js
+assert(document.querySelectorAll('label')[3].matches('label[for="new-password"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8b9dab5ac88e4d3d43a3.md:61]:
```suggestion
assert(document.querySelectorAll('label')?.[3]?.matches('label[for="new-password"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
+---
+id: 62ff8e998d3e7eae14d6ae3b
+title: Step 28
+challengeType: 0
+dashedName: step-28
+---
+
+# --description--
+
+Now, to follow accessiblity best practices, link the `input` elements and the `label` elements in the second fieldset together using the `for` attribute.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:10]:
```suggestion
Follow accessibility best practices by linking the `input` elements and the `label` elements in the second `fieldset`.
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+Use these values in order for the `id`s: `personal-account`, `business-account`, `terms-and-conditions`
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:12]:
```suggestion
Use these values for the respective `id` properties: `personal-account`, `business-account`, `terms-and-conditions`
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+# --hints--
+
+The first `input` element should have an `id` of `personal-account`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(2) input')[0].matches('#personal-account'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:19]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(2) input')?.[0]?.matches('#personal-account'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The second `input` element should have an `id` of `business-account`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(2) input')[1].matches('#business-account'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:25]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(2) input')?.[1]?.matches('#business-account'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The third `input` element should have an `id` of `terms-and-conditions`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(2) input')[2].matches('#terms-and-conditions'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:31]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(2) input')?.[2]?.matches('#terms-and-conditions'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The first `label` element should have a `for` attribute with a value of `personal-account`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(2) label')[0].matches('label[for="personal-account"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:37]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(2) label')?.[0]?.matches('label[for="personal-account"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The second `label` element should have a `for` attribute with a value of `business-account`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(2) label')[1].matches('label[for="business-account"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:43]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(2) label')?.[1]?.matches('label[for="business-account"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The third `label` element should have a `for` attribute with a value of `terms-and-conditions`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(2) label')[2].matches('label[for="terms-and-conditions"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff8e998d3e7eae14d6ae3b.md:49]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(2) label')?.[2]?.matches('label[for="terms-and-conditions"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 20]
```diff
@@ -0,0 +1,131 @@
+---
+id: 62ff919a7b5612c0670923a5
+title: Step 37
+challengeType: 0
+dashedName: step-37
+---
+
+# --description--
+
+Now, to follow accessiblity best practices, link the form elements and their `label` elements together using the `for` attribute.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:10]:
```suggestion
Link the applicable form elements and their `label` elements together.
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 21]
```diff
@@ -0,0 +1,131 @@
...
+
+Use these values in order for the `id`s: `profile-picture`, `age`, `referrer`, `bio`
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:12]:
```suggestion
Use these values for the respective `id` properties: `profile-picture`, `age`, `referrer`, `bio`
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+
+# --hints--
+
+The first `input` element should have an `id` of `profile-picture`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(3) input')[0].matches('#profile-picture'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:19]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) input')?.[0]?.matches('#profile-picture'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 23]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+The second `input` element should have an `id` of `age`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(3) input')[1].matches('#age'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:25]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) input')?.[1]?.matches('#age'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 24]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+The `select` element should have an `id` of `referrer`.
+
+```js
+assert(document.querySelector('fieldset:nth-of-type(3) select').matches('#referrer'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:31]:
```suggestion
assert(document.querySelector('fieldset:nth-of-type(3) select')?.matches('#referrer'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 25]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+The `textarea` element should have an `id` of `bio`.
+
+```js
+assert(document.querySelector('fieldset:nth-of-type(3) textarea').matches('#bio'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:37]:
```suggestion
assert(document.querySelector('fieldset:nth-of-type(3) textarea')?.matches('#bio'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 26]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+The first `label` element should have a `for` attribute with a value of `profile-picture`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[0].matches('label[for="profile-picture"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:43]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[0]?.matches('label[for="profile-picture"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 27]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+The second `label` element should have a `for` attribute with a value of `age`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[1].matches('label[for="age"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:49]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[1]?.matches('label[for="age"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 28]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+The third `label` element should have a `for` attribute with a value of `referrer`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[2].matches('label[for="referrer"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:55]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[2]?.matches('label[for="referrer"]'))
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md [hunk 29]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+The fourth `label` element should have a `for` attribute with a value of `bio`.
+
+```js
+assert(document.querySelectorAll('fieldset:nth-of-type(3) label')[3].matches('label[for="bio"]'))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/62ff919a7b5612c0670923a5.md:61]:
```suggestion
assert(document.querySelectorAll('fieldset:nth-of-type(3) label')?.[3]?.matches('label[for="bio"]'))
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you for your patience with me getting to this. Also, thank you for teaching me about the `.matches` method.

Ideally, we would use specific assertions, but that is most _out the window_ now.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

**Pull request merged**